app/src/main/java/com/example/foody/recipe_search/presentation_mvvm/ui/SearchScreen.kt

```diff
@@ -173,43 +217,43 @@ private fun RecipeItem(item: RecipeInfo, goToDetailsScreen: (String) -> Unit) {
 
 @Composable
 private fun SearchTextFieldAndButton(
+    modifier: Modifier = Modifier,
     state: SearchScreenState,
     performSearch: () -> Unit,
-    onValueChanged: (searchTerm: String) -> Unit
+    onValueChanged: (searchTerm: String) -> Unit,
 ) {
     Row(
-        modifier = Modifier.padding(horizontal = 8.dp, vertical = 8.dp),
+        modifier = modifier.padding(horizontal = 8.dp, vertical = 16.dp),
         verticalAlignment = Alignment.CenterVertically
     ) {
         TextField(
-            value = state.searchTerm,
+            value = state.searchBarState.searchTerm,
             onValueChange = { onValueChanged.invoke(it) }, // updates the changes on searchTerm
-            modifier = Modifier
+            modifier = modifier
                 .height(48.dp)
                 .weight(1f)
                 .clip(shape = RoundedCornerShape(12.dp))
         )
-        Spacer(modifier = Modifier.size(16.dp))
-        Button(onClick = { performSearch() }) { Text(text = "Search") }
+        Spacer(modifier = modifier.size(16.dp))
+        Button(onClick = performSearch) {
+            Text(text = "Search")
+        }
     }
 }
 
 @Composable
-private fun RecipeTitle(title: String) {
+private fun RecipeTitle(modifier: Modifier = Modifier, title: String) {
     Text(
         text = title,
         textAlign = TextAlign.Center,
         style = MaterialTheme.typography.titleMedium,
         fontWeight = FontWeight.Bold,
-        modifier = Modifier
+        modifier = modifier
```

Comment:
In case when we adding some "override" or "default" functionality (via Modifier), do not add it directly to the passed in modifier. Chain it with, either `modifier.then(Modifier.someApplications())`, or `Modifier.someApplications().then(modifier)`. So not directly just `modifier.someApplications()`.

Reply:
Like in my examples [here](https://github.com/FilipM79/Foody/pull/5#discussion_r1534135858)